lib/eventq/eventq_aws/aws_queue_client.rb

```diff
@@ -4,29 +4,45 @@ module EventQ
   module Amazon
     class QueueClient
       def initialize(options = {})
-        invalid_keys = options.keys - [:sns_keep_alive_timeout, :sns_continue_timeout ]
+        invalid_keys = options.keys - %i[sns_keep_alive_timeout sns_continue_timeout]
         raise(OptionParser::InvalidOption, invalid_keys) unless invalid_keys.empty?
 
         @sns_keep_alive_timeout = options[:sns_keep_alive_timeout] || 30
         @sns_continue_timeout = options[:sns_continue_timeout] || 15
       end
 
       # Returns the AWS SQS Client
-      def sqs
-        @sqs ||= sqs_client
+      def sqs(region = nil)
+        if region.nil?
+          @sqs ||= sqs_client
+        else
+          sqs_client(region)
+        end
       end
 
       # Returns the AWS SNS Client
-      def sns
-        @sns ||= sns_client
+      def sns(region = nil)
+        if region.nil?
+          @sns ||= sns_client
+        else
+          sns_client(region)
+        end
       end
 
-      def sqs_helper
-        @sqs_helper ||= Amazon::SQS.new(sqs)
+      def sqs_helper(region = nil)
+        if region.nil?
+          @sqs_helper ||= Amazon::SQS.new(sqs)
+        else
+          Amazon::SQS.new(sqs_client(region))
+        end
       end
 
-      def sns_helper
-        @sns_helper ||= Amazon::SNS.new(sns)
+      def sns_helper(region = nil)
+        if region.nil?
```

Comment:
I don't think we need this nil guard because `sns_client` deals with region being nil or not

Reply:
ah so this nil guard is because for normal usage (default region) the client is memoized, to make calls to a different region requires a new instance of a client. So we need this nil guard otherwise we would have to drop the memoization of the client.